lib/active_record/connection_adapters/trilogy_adapter.rb

```diff
@@ -153,13 +153,99 @@ def discard!
         self.connection = nil
       end
 
+      # ActiveRecord 7.0 support
+      if ActiveRecord.version < ::Gem::Version.new('7.1.a')
+        def initialize(*args, **kwargs)
+          if kwargs.present?
+            args << kwargs.dup
+            kwargs.clear
+          end
+          # Turn  .new(config)  into  .new(nil, nil, nil, config)
+          3.times { args.unshift nil } if args.length < 4
+          super
```

Comment:
since this constructor is only called for AR 7.0 which expects arguments in the form of 
```ruby
initialize(connection, logger, connection_options, config)
```

so I don't think that it's possible for the adapter to be initialized with just `config` outside of tests. Which if that's the case IMO we should change the tests based on the AR version.